torch/testing/_internal/common_mkldnn.py

```diff
@@ -0,0 +1,85 @@
+import contextlib
+import functools
+import inspect
+
+import torch
+
+
+# Test whether hardware BF32 math mode enabled. It is enabled only on:
+# - MKLDNN is available
+# - BF16 is supported by MKLDNN
+def bf32_is_not_fp32():
+    if not torch.backends.mkldnn.is_available():
+        return False
+    if not torch.ops.mkldnn._is_mkldnn_bf16_supported():
+        return False
+    return True
+
+
+@contextlib.contextmanager
+def bf32_off():
+    try:
+        with torch.backends.mkldnn.flags(enabled=None, allow_bf32=False):
+            yield
+    finally:
+        pass
+
+
+@contextlib.contextmanager
+def bf32_on(self, bf32_precision=1e-5):
```

Comment:
May I know how `bf32_precision` is applied to the test?

Reply:
Yes, with this https://github.com/pytorch/pytorch/pull/116015/files/f6f4686015a4a8795fd5305473cc232db79abc4c#diff-7e17421f32124016eb8de04dc2f445da5786a28355e1addc72b305466f590180R609 will run bf32_on (following `with_tf32_on_and_off`)

Reply:
This doesn't seem correct. `tf32_on` changes `self.precision` per `tf32_precision` arg (https://github.com/pytorch/pytorch/blob/e3aefe297050e197ff3f36f555a134a0e61ffc9c/torch/testing/_internal/common_cuda.py#L94). But you don't. Why?

Reply:
Thanks for the reminder! @jgong5 , I misunderstood your question before. Now I have fixed it.